### PR TITLE
[antlr4] update to 4.9.3

### PR DESCRIPTION
**Pull request by niyue**
**Describe the pull request**

- #### What does your PR fix?  
It upgrades antlr4 from 4.9.1 to 4.9.3

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes

### CODE REVIEW

**Comment by niyue** [comment]
It looks like there are some CI build failures on Windows. I don't have a Window env to verify the PR and only have verified this PR on macOS so far. 

I checked out this CI check:
https://github.com/microsoft/vcpkg/pull/22438/checks?check_run_id=4752639338

And it says the detailed log is available at `D:\buildtrees\antlr4\config-x86-windows-out.log`, could anyone let me know where I can get this log from CI? Thanks.

File: ports/antlr4/portfile.cmake [hunk 1]
```diff
@@ -24,7 +24,6 @@ vcpkg_extract_source_archive_ex(
         fixed_build.patch
         uuid_discovery_fix.patch
         export_guid.patch
-        fix_utfcpp_dependency.patch
 )
 
 vcpkg_configure_cmake(
```
Comment (JonLiu1993) [inline ports/antlr4/portfile.cmake:29]: @niyue ,Thanks for your pr, could you modernize these cmake functions?

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
> It looks like there are some CI build failures on Windows. I don't have a Window env to verify the PR and only have verified this PR on macOS so far.
> 
> I checked out this CI check: https://github.com/microsoft/vcpkg/pull/22438/checks?check_run_id=4752639338
> 
> And it says the detailed log is available at `D:\buildtrees\antlr4\config-x86-windows-out.log`, could anyone let me know where I can get this log from CI? Thanks.

@niyue ,You can view the CI detailed error log through the link below:
https://dev.azure.com/vcpkg/public/_build/results?buildId=65569&view=artifacts&pathAsName=false&type=publishedArtifacts

File: ports/antlr4/portfile.cmake (hunk 1 above)
Comment (niyue) [reply ports/antlr4/portfile.cmake:29]: Sure, let me give it a try.

**Comment by niyue** [comment]
@JonLiu1993 I downloaded `failure logs for x64-windows` and found it reports error like below:
```
-- Output libraries to D:/buildtrees/antlr4/x64-windows-rel/dist
CMake Error at D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/ExternalProject.cmake:2650 (message):
  error: could not find git for clone of utf8cpp
Call Stack (most recent call first):
  D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/ExternalProject.cmake:3700 (_ep_add_download_command)
  runtime/CMakeLists.txt:58 (ExternalProject_Add)
```
It looks like cmake complains `git` is not available in this environment when cloning `utf8cpp`. Since I don't touch anything related with `git` or add any new dependency in this PR, is this error expected (probably this env should have `git` installed)? 4 other environments in the CI don't report such an error.

File: ports/antlr4/portfile.cmake (hunk 1 above)
Comment (niyue) [reply ports/antlr4/portfile.cmake]: Done

File: ports/antlr4/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "antlr4",
-  "version-string": "4.9.1",
+  "version-string": "4.9.3",
```
[the remark this answers is not in the document]
Comment (niyue) [reply ports/antlr4/vcpkg.json]: Done

File: ports/antlr4/vcpkg.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
   "port-version": 1,
```
[the remark this answers is not in the document]
Comment (niyue) [reply ports/antlr4/vcpkg.json]: Removed.

**Comment by dg0yt** [comment]
I think it shouldn't fetch or `git clone` anything once entering CMake project mode. It will break offline building.

**Comment by niyue** [comment]
> I think it shouldn't fetch or `git clone` anything once entering CMake project mode. It will break offline building.

Is offline building a requirement for a port? Where can I find documentation about this requirement? Is this a new requirement for port verification (since 4.9.1 seems to work and it should do the same thing using git clone). I don't change anything about this part, and just want to upgrade antlr from 4.9.1 to 4.9.3.

**Comment by dg0yt** [comment]
> > I think it shouldn't fetch or `git clone` anything once entering CMake project mode. It will break offline building.
> 
> Is offline building a requirement for a port? Where can I find documentation about this requirement? Is this a new requirement for port verification (since 4.9.1 seems to work and it should do the same thing using git clone). I don't change anything about this part, and just want to upgrade antlr from 4.9.1 to 4.9.3.

Offline building is a requirement for some (usually corporate) users. And hidden network access sometimes breaks CI because it circumvents the regular asset caching. 

In addition, it indicates "vendored dependencies". Read: https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-use-vendored-dependencies

4.9.1 of this port carried a patch to remove the vendored utfcpp. This patch may need updating, not removing it.

**Comment by niyue** [comment]
> In addition, it indicates "vendored dependencies". Read: https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-use-vendored-dependencies
> 
> 4.9.1 of this port carried a patch to remove the vendored utfcpp. This patch may need updating, not removing it.

Thanks so much for the detailed information. Let me examine the patch again.

**Comment by niyue** [comment]
@JonLiu1993 I managed to pass all CI checks (except the `PR suggestions` check, which seems a Github action network failure, and it is not related with this PR, and it passed in previous CI runs), could you please help to check it out? Thanks.

**Comment by niyue** [comment]
@JonLiu1993 I added some minor updates and trigger the CI to run again and it passed all its checks now. Could you please check it out? Thanks.

File: ports/antlr4/vcpkg.json [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
       "name": "libuuid",
       "platform": "!uwp & !windows & !osx"
+    },
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
```
Comment (JonLiu1993) [inline ports/antlr4/vcpkg.json:18]:
```suggestion
```

Comment (JonLiu1993) [reply ports/antlr4/vcpkg.json:18]: This dependency is not actually needed

**Review by JonLiu1993 — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**